pageserver/src/tenant/timeline.rs

```diff
@@ -1290,7 +1370,7 @@ impl Timeline {
                     updates.insert_historic(remote_layer);
                 }
                 #[cfg(test)]
-                LayerFileName::Test(_) => unreachable!(),
+                LayerFileName::Test(_) => unimplemented!(),
```

Comment:
I think this is actually unreachable, the tests never exercise this code path.

Reply:
I plan to remove the entire variant after this PR lands.
I've actually done that in this PR after all moves, but reverted that.

Reply:
Yeah, saw that, but let's do that in a separate PR

Reply:
Sure, that's what I wanted.
My point here is that let's just leave it as is and it'll be gone in the next PR hopefully.

Reply:
Since had to push more code, changed this back to `unreachable!`